lib/db_utils.go

```diff
@@ -10661,6 +10681,409 @@ func DBDeletePostAssociationWithTxn(txn *badger.Txn, snap *Snapshot, association
+			txn, snap, profilePKID, lockupDurationNanoSecs)
+		return nil
+	})
+	return leftLockupYieldCurvePoint, rightLockupYieldCurvePoint
+}
+
+func DBGetLeftLockupYieldCurvePointWithTxn(txn *badger.Txn, snap *Snapshot, profilePKID *PKID,
+	lockupDurationNanoSecs int64) (_leftLockupYieldCurvePoint *LockupYieldCurvePoint) {
+	key := _dbKeyForLockupYieldCurvePoint(LockupYieldCurvePoint{
+		ProfilePKID:            profilePKID,
+		LockupDurationNanoSecs: lockupDurationNanoSecs,
+	})
+
+	// Seek left of the yield curve point.
+	iterLeftOpts := badger.DefaultIteratorOptions
+	iterLeftOpts.Reverse = true
+	iterLeft := txn.NewIterator(iterLeftOpts)
+	iterLeft.Seek(key)
+	iterLeftKey := iterLeft.Item().Key()
+
+	// There's a chance our seek yield a key in a different prefix (i.e. not a yield curve point).
+	// In this case, we know _dbKeyToLockupYieldCurvePoint will fail in parsing the key.
+	// We can return early in this case as there's no relevant yield points in the DB.
+	if len(iterLeftKey) < len(Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs) {
+		return nil
+	}
+	if !bytes.Equal(iterLeftKey[:len(Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs)],
+		Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs) {
+		return nil
+	}
+
+	// Fetch the LockupYieldCurvePoint beneath the key.
+	leftLockupYieldCurvePointBytes, err := DBGetWithTxn(txn, snap, iterLeftKey)
+	if err != nil {
+		return nil
+	}
+
+	// Parse the bytes beneath the key.
+	leftLockupYieldCurvePointObj := &LockupYieldCurvePoint{}
+	rr := bytes.NewReader(leftLockupYieldCurvePointBytes)
+	DecodeFromBytes(leftLockupYieldCurvePointObj, rr)
+	return leftLockupYieldCurvePointObj
+}
+
+func DBGetRightLockupYieldCurvePointWithTxn(txn *badger.Txn, snap *Snapshot, profilePKID *PKID,
+	lockupDurationNanoSecs int64) (_rightLockupYieldCurvePoint *LockupYieldCurvePoint) {
+	key := _dbKeyForLockupYieldCurvePoint(LockupYieldCurvePoint{
+		ProfilePKID:            profilePKID,
+		LockupDurationNanoSecs: lockupDurationNanoSecs,
+	})
+
+	// Seek left of the yield curve point.
+	iterRightOpts := badger.DefaultIteratorOptions
+	iterRight := txn.NewIterator(iterRightOpts)
+	iterRight.Seek(key)
+	iterRightKey := iterRight.Item().Key()
+
+	// There's a chance our seek yield a key in a different prefix (i.e. not a yield curve point).
+	// In this case, we know _dbKeyToLockupYieldCurvePoint will fail in parsing the key.
+	// We can return early in this case as there's no relevant yield points in the DB.
+	if len(iterRightKey) < len(Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs) {
+		return nil
+	}
+	if !bytes.Equal(iterRightKey[:len(Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs)],
+		Prefixes.PrefixLockupYieldCurvePointByProfilePKIDAndDurationNanoSecs) {
+		return nil
+	}
+
+	// Fetch the LockupYieldCurvePoint beneath the key.
+	rightLockupYieldCurvePointBytes, err := DBGetWithTxn(txn, snap, iterRightKey)
```

Comment:
The same bug I mentioned for the left seeker applies here. It could return a point for another user.